src/app/components/login/login.component.ts

```diff
@@ -48,11 +48,20 @@ export class LoginComponent implements OnInit {
         sessionStorage.setItem('currentUser', JSON.stringify(currentUser));
         console.log('Now going to homepage...');
         this.loginService.storeManagerIdFromServer(currentUser.email)
+        this.sleep(500);
```

Comment:
Why are we hardcoding this sleep here? Can you guarantee that the `this.loginService.storeManagerIdFromServer` will resolve before the timer completes?

Consider the following approach:
- return the observable from `loginService#storeManagerIdFromServer`
- In `login.component.ts`, do something like the following
```typescript
this.loginService.storeManagerIdFromServer(currentUser.email).pipe(
  map(() => this.router.navigate(['home'])),
  catchError(() => this.router.navigate(['login']))
).subscribe(didRoute => {
 console.log("Did we route? " + didRoute);
});
```